sw/banshee/src/tran.rs

```diff
@@ -2095,1423 +2470,4496 @@ impl<'a> InstructionTranslator<'a> {
-            riscv::OpcodeRdRs1Rs2::FsgnjnS => {
-                self.write_freg_f32(
-                    data.rd,
-                    self.emit_fsgnjn(self.read_freg_f32(data.rs1), self.read_freg_f32(data.rs2)),
+                let reg_masked = LLVMBuildAnd(
+                    self.builder,
+                    reg,
+                    LLVMConstInt(LLVMInt32Type(), 0xf8 as u64, 0),
+                    NONAME,
                 );
-                return Ok(());
-            }
-            riscv::OpcodeRdRs1Rs2::FsgnjxS => {
-                self.write_freg_f32(
-                    data.rd,
-                    self.emit_fsgnjx(self.read_freg_f32(data.rs1), self.read_freg_f32(data.rs2)),
+                let rs2_dm = LLVMBuildAnd(
+                    self.builder,
+                    rs2,
+                    LLVMConstInt(LLVMInt32Type(), 0x1f as u64, 0),
+                    NONAME,
                 );
-                return Ok(());
-            }
-            riscv::OpcodeRdRs1Rs2::FsgnjD => {
-                self.write_freg_f64(
-                    data.rd,
-                    self.emit_fsgnj(self.read_freg_f64(data.rs1), self.read_freg_f64(data.rs2)),
+                let rs2_dm_shifted = LLVMBuildShl(
+                    self.builder,
+                    rs2_dm,
+                    LLVMConstInt(LLVMInt32Type(), 8 as u64, 0),
+                    NONAME,
                 );
-                return Ok(());
-            }
-            riscv::OpcodeRdRs1Rs2::FsgnjnD => {
-                self.write_freg_f64(
-                    data.rd,
-                    self.emit_fsgnjn(self.read_freg_f64(data.rs1), self.read_freg_f64(data.rs2)),
+                let addr_off = LLVMBuildOr(self.builder, rs2_dm_shifted, reg_masked, NONAME);
+
+                let addr = LLVMBuildAdd(
+                    self.builder,
+                    LLVMConstInt(LLVMInt32Type(), SSR_BASE, 0),
+                    addr_off,
+                    NONAME,
                 );
+                self.write_mem(addr, rs1, 2);
                 return Ok(());
             }
-            riscv::OpcodeRdRs1Rs2::FsgnjxD => {
-                self.write_freg_f64(
-                    data.rd,
-                    self.emit_fsgnjx(self.read_freg_f64(data.rs1), self.read_freg_f64(data.rs2)),
+            _ => (),
+        };
+
+        match data.op {
+            riscv::OpcodeRs1Rs2::Dmsrc => self
+                .section
+                .emit_call("banshee_dma_src", [self.dma_ptr(), rs1, rs2]),
+            riscv::OpcodeRs1Rs2::Dmdst => self
+                .section
+                .emit_call("banshee_dma_dst", [self.dma_ptr(), rs1, rs2]),
+            riscv::OpcodeRs1Rs2::Dmstr => self
+                .section
+                .emit_call("banshee_dma_str", [self.dma_ptr(), rs1, rs2]),
+            _ => bail!("Unsupported opcode {}", data.op),
+        };
+        Ok(())
+    }
+
+    unsafe fn emit_unit(&self, data: riscv::FormatUnit) -> Result<()> {
```

Comment:
The diff becomes huge here, but this is a Github issue, right? Or did you change the `emit_unit` function here?

Reply:
This should be a Github issue. Afaik this function was not changed